posthog/api/plugin.py

```diff
@@ -91,7 +106,15 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
     def update(self, plugin: Plugin, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:  # type: ignore
         if not can_install_plugins_via_api(self.context["organization_id"]):
             raise ValidationError("Plugin upgrades via the web are disabled!")
-        if plugin.plugin_type != Plugin.PluginType.SOURCE:
+        if (
+            validated_data.get("is_global")
+            and Organization.objects.only("plugins_access_level")
+            .get(id=self.context["organization_id"])
+            .plugins_access_level
+            < Organization.PluginsAccessLevel.ROOT
+        ):
+            raise PermissionError("This organization can't manage global plugins!")
+        if not validated_data and plugin.plugin_type != Plugin.PluginType.SOURCE:
             validated_data = self._update_validated_data_from_url({}, plugin.url)
```

Comment:
We probably need to split this into two actions: 1) updating the is_global prop, 2) updating to the latest plugin version